dmpy/client.py

```diff
@@ -0,0 +1,147 @@
+import hashlib
+import json
+import logging
+from datetime import datetime
+from pathlib import Path
+
+import requests
+from dotenv import get_key, load_dotenv, set_key
+from requests_toolbelt.multipart.encoder import (
+    MultipartEncoder,
+    MultipartEncoderMonitor,
+)
+
+from .core.payloads import FileUploadPayload
+from .core.utils import read_text_resource
+
+log = logging.getLogger(__name__)
+
+
+class Dmpy:
+    def __init__(self):
+        self.env = ".dmpy.env"
+        self.url = get_key(self.env, "DMP_URL")
+        load_dotenv(self.env)
+
+    def access_token(self) -> str:
+        """Obtain (or refresh) an access token."""
+        now = int(datetime.utcnow().timestamp())
+        last_created = int(get_key(self.env, "DMP_ACCESS_TOKEN_GEN_TIME"))
```

Comment:
Is the `Dmpy` object kept alive (which seems efficient anyways)? If so - we could also consider not storing this in the `.env`, but instead in the class `.self` variable. If not - any thoughts on instantiating one at a higher level for using across uploads?

Reply:
Currently it is not kept alive; if two instances are created then the same access token would be used unless the predetermined time has passed. This would not be the case unless stored in a way that can be shared across instances. An alternative (perhaps simplier) solution would be to store the data in a variable outside of the class (above it) which then becomes "static" due to python scoping.

This `access_token` method should probs be "private" too given it's only used in `upload` and shouldn't be used publically?

Reply:
Just for clarification, does that mean that for each outgoing request we instantiate a new `Dmpy` object? 
I'd leave this for now, yes

Reply:
Sounds like a decision for the middleware 👍🏼 

(**Yes** -- we currently create a new instance of [Dmpy with each upload request](https://github.com/ideafast/middleware-services/blob/ca56f0e500b89d7a7a2feb0e750702b9522ef7b9/data_transfer/services/dmpy.py#L41))